app/scripts/controllers/transactions/index.js

```diff
@@ -793,4 +793,9 @@ export default class TransactionController extends EventEmitter {
     const currentNetworkTxList = this.txStateManager.getTxList(MAX_MEMSTORE_TX_LIST_SIZE)
     this.memStore.updateState({ unapprovedTxs, currentNetworkTxList })
   }
+
+  clearUnapprovedTxs () {
+    this.memStore.updateState({ unapprovedTxs: {}, currentNetworkTxList: [] })
+    this.txStateManager.clearTxs()
```

Comment:
Would this be equivalent?
```suggestion
    this.txStateManager.clearTxs()
```

This feels more appropriate, since `unapprovedTxs` and `currentNetworkTxList` are both derived from `txStateManager`, and `txStateManager` updates trigger these to update in turn (via `_updateMemstore`, which is triggered by any `txStateManager` update)